packages/dd-trace/src/appsec/api_security_sampler.js

```diff
@@ -1,61 +1,77 @@
 'use strict'
 
+const TTLCache = require('@isaacs/ttlcache')
+const web = require('../plugins/util/web')
 const log = require('../log')
+const { AUTO_REJECT, USER_REJECT } = require('../../../../ext/priority')
 
-let enabled
-let requestSampling
+const MAX_SIZE = 4096
 
-const sampledRequests = new WeakSet()
+let enabled
+let sampledRequests
 
 function configure ({ apiSecurity }) {
   enabled = apiSecurity.enabled
-  setRequestSampling(apiSecurity.requestSampling)
+  sampledRequests = new TTLCache({ max: MAX_SIZE, ttl: apiSecurity.sampleDelay * 1000 })
 }
 
 function disable () {
   enabled = false
+  sampledRequests?.clear()
 }
 
-function setRequestSampling (sampling) {
-  requestSampling = parseRequestSampling(sampling)
-}
+function sampleRequest (req, res, force = false) {
+  if (!enabled) return false
+  if (isSampled(req, res)) return false
```

Comment:
bad copy paste ? :raised_eyebrow:

Reply:
I had some conflicts seems I duplicate the code